diff --git a/ios/Core/ExpoBridgeModule.mm b/ios/Core/ExpoBridgeModule.mm
index 2ed1c00f47406e109750cc27ace7e0d88e42c00e..d14269aae847143318888ad3c848d707de95e691 100644
--- a/ios/Core/ExpoBridgeModule.mm
+++ b/ios/Core/ExpoBridgeModule.mm
@@ -45,9 +45,9 @@ - (void)setBridge:(RCTBridge *)bridge
   _bridge = bridge;
   _appContext.reactBridge = bridge;
 
-#if !__has_include(<ReactCommon/RCTRuntimeExecutor.h>)
-  _appContext._runtime = [EXJavaScriptRuntimeManager runtimeFromBridge:bridge];
-#endif // React Native <0.74
+// #if !__has_include(<ReactCommon/RCTRuntimeExecutor.h>)
+//   _appContext._runtime = [EXJavaScriptRuntimeManager runtimeFromBridge:bridge];
+// #endif // React Native <0.74
 }
 
 #if __has_include(<ReactCommon/RCTRuntimeExecutor.h>)
